### PR TITLE
[qt6] A few more QRegExp aurevoirs in src/core

### PR DESCRIPTION
**Pull request by nirvn**
## Description

A few more steps in the purging of QRegExp.

### CODE REVIEW

**Comment by nirvn** [comment]
OK, src/core/expression migrated away from QRegExp too. In theory, it should make expression parsing faster too.

One note that might help others: QRegExp::escape( QString ) != QRegularExpression::escape( QString ). The deprecated QRegExp's escape function escapes a narrow number of characters. There's an easy work around that's used here which can be replicated elsewhere if needed.

File: src/core/expression/qgsexpressionnodeimpl.cpp [hunk 1]
```diff
@@ -531,39 +531,42 @@ QVariant QgsExpressionNodeBinaryOperator::evalNode( QgsExpression *parent, const
...
         QString regexp = QgsExpressionUtils::getStringValue( vR, parent );
         ENSURE_NO_EVAL_ERROR
-        // TODO: cache QRegExp in case that regexp is a literal string (i.e. it will stay constant)
+        // TODO: cache QRegularExpression in case that regexp is a literal string (i.e. it will stay constant)
         bool matches;
         if ( mOp == boLike || mOp == boILike || mOp == boNotLike || mOp == boNotILike ) // change from LIKE syntax to regexp
         {
-          QString esc_regexp = QRegExp::escape( regexp );
+          // recreate QRegExp::escape function behavior
+          const thread_local QRegularExpression escapeRx( QStringLiteral( "([^\\\\])([\\$\\(\\)\\*\\+\\.\\?\\[\\]\\^\\{\\|\\}])" ) );
```
[the remark this answers is not in the document]
Comment (nirvn) [reply src/core/expression/qgsexpressionnodeimpl.cpp:539]: @rouault , fair point. The updated logic initially failed (while the expression tests themselves were fine, the app locator tested one type of ilike which got broken). I'll add a few additional tests tomorrow.

**Comment by nirvn** [comment]
OK a few more LIKE expression tests were added, including the particular string that failed in the app locator test. In the process, I've uncovered another issue with the QRegExp::escape replacement. 

I feel pretty sure our bases are covered now.

**Review by rouault — APPROVED** [review]

**Comment by agiudiceandrea** [comment]
Hi @nirvn, it seems to me this PR introduced a regression for LIKE / ILIKE operators and text containing line breaks.
Could you please have a look at the issue report https://github.com/qgis/QGIS/issues/47957 and at my proposed fix https://github.com/qgis/QGIS/pull/50124?

**Pull request merged**